mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"time"
+)
+
+type InactiveReviewHandler struct{}
+
+func init() {
+	h := &InactiveReviewHandler{}
+	RegisterMungerOrDie(h)
+}
+
+// Name is the name usable in --pr-mungers
+func (*InactiveReviewHandler) Name() string { return "inactive-review-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (*InactiveReviewHandler) RequiredFeatures() []string {
+	return []string{}
+}
+
+// Initialize will initialize the munger
+func (*InactiveReviewHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (*InactiveReviewHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (*InactiveReviewHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+func (*InactiveReviewHandler) suggestNewReviewer(obj *github.MungeObject, feats *features.Features) string {
+	issue := obj.Issue
+
+	//do not suggest new reviewer if the PR has no author information
+	if issue == nil || issue.PullRequestLinks == nil || issue.User == nil || issue.User.Login == nil {
+		return ""
+	}
+
+	files, ok := obj.ListFiles()
+
+	if !ok || files == nil || len(files) == 0 {
+		return ""
+	}
+
+	potentialOwners, weightSum := getPotentialOwnersHelper(issue.User.Login, feats, files, issue.Number)
+
+	if len(potentialOwners) > 0 && issue.Assignees != nil && len(issue.Assignees) > 0 {
+		oldReviewers := issue.Assignees
+
+		for _, oldReviewer := range oldReviewers {
+			login := *oldReviewer.Login
+
+			for potentialOwner := range potentialOwners {
+				if login == potentialOwner {
+					weightSum -= potentialOwners[login]
+					delete(potentialOwners, login)
+					break
+				}
+			}
+		}
+	}
+
+	if len(potentialOwners) == 0 {
```

Comment:
`len(potentialOwners)` will change as there is a line `delete(potentialOwners, login)`. If i move the check `if len(potentialOwners) == 0` before `if len(potentialOwners) > 0 && ...`, i will have to check it again afterwards.